File: Algorithms/InsertionSort/insertion_sort.py
"""
Pseudocode
for i = 1 to n
    key = A[i]
    j = i - 1
    while j >= 0 and A[j] > key
        A[j+1] = A[j]
        j = j - 1
    End while
    A[j+1] = key
End for
"""

def insertion_sort(arr):
    for i in range(1, len(arr)):  # running from 1 index, 0 already sorted
        key = arr[i]              # 2nd elements of an array
        j = i-1                   # back to the first element to check (first, next iter)
        while j >= 0 and (arr[j] > key):
            # moving the right side element to the one position backword
            # until it reaches to the correct position
            arr[j+1] = arr[j]
            j -= 1
        # moving current element to its currect position
        arr[j+1] = key
    return arr


if __name__ == "__main__":
    arr = [4, 5, 6, 2, 1, 9]
    print("Sorted array using Insertion sort: ", insertion_sort(arr))

File: Algorithms/InsertionSort/insertion_sort.js
insertion_sort = (arr) => {
    for(let i = 1; i < arr.length; i++){
        // get the 2nd index value 
        let key = arr[i];
        // first index value of array
        let j = i-1;
        while (j >= 0 && arr[j] > key){
            // moving the right side element to one position backword, 
            // until it reaches to the correct position
            arr[j+1] = arr[j];
            j--;
        }
        // moving the last element to its correct position
        arr[j+1] = key
    }
    console.log("Sorted array using Insertion Sort: ",arr)
}

arr = [6, 5, 7, 3, 1, 9]
insertion_sort(arr)